pkg/sentry/socket/epsocket/stack.go

```diff
@@ -144,7 +144,98 @@ func (s *Stack) SetTCPSACKEnabled(enabled bool) error {
 
 // Statistics implements inet.Stack.Statistics.
 func (s *Stack) Statistics(stat interface{}, arg string) error {
-	return syserr.ErrEndpointOperation.ToError()
+	switch stats := stat.(type) {
+	case *inet.StatSNMPIP:
+		ip := Metrics.IP
+		*stats = inet.StatSNMPIP{
+			0, // Forwarding
+			0, // DefaultTTL
+			ip.PacketsReceived.Value(), // InReceives
+			0, // InHdrErrors
+			ip.InvalidAddressesReceived.Value(), // InAddrErrors
+			0, // ForwDatagrams
+			0, // InUnknownProtos
+			0, // InDiscards
+			ip.PacketsDelivered.Value(),     // InDelivers
+			ip.PacketsSent.Value(),          // OutRequests
+			ip.OutgoingPacketErrors.Value(), // OutDiscards
+			0, // OutNoRoutes
+			0, // ReasmTimeout
+			0, // ReasmReqds
+			0, // ReasmOKs
+			0, // ReasmFails
+			0, // FragOKs
+			0, // FragFails
+			0, // FragCreates
+		}
+	case *inet.StatSNMPICMP:
+		in := Metrics.ICMP.V4PacketsReceived.ICMPv4PacketStats
+		out := Metrics.ICMP.V4PacketsSent.ICMPv4PacketStats
+		*stats = inet.StatSNMPICMP{
+			0, // InMsgs
+			Metrics.ICMP.V4PacketsSent.Dropped.Value(), // InErrors
+			0, // InCsumErrors
+			in.DstUnreachable.Value(), // InDestUnreachs
+			in.TimeExceeded.Value(),   // InTimeExcds
+			in.ParamProblem.Value(),   // InParmProbs
+			in.SrcQuench.Value(),      // InSrcQuenchs
+			in.Redirect.Value(),       // InRedirects
+			in.Echo.Value(),           // InEchos
+			in.EchoReply.Value(),      // InEchoReps
+			in.Timestamp.Value(),      // InTimestamps
+			in.TimestampReply.Value(), // InTimestampReps
+			in.InfoRequest.Value(),    // InAddrMasks
+			in.InfoReply.Value(),      // InAddrMaskReps
+			0,                         // OutMsgs
```

Comment:
Is OutMsgs not available in Netstack?

Reply:
I don't find a proper one. Any suggestion?

Reply:
I think this is fine for now but best create a bug to implement it.